### ISSUE TITLE
Documentation: git repros with submodules should never be moved with older versions of git

### ISSUE THREAD

**Issue by hamogu** [astropy/astropy#2646]
I have run into the following issue and I believe we shuld add a warning to the developer documentation of astropy and all packages that make use of `astropy-helpers` (incl. `astropy-template`). I just don't know where a good place for such a warning would be.

This is the issue:
After `astropy-helpers` came into effect, I moved the directory where I develop astropy to a different location. 
The results is this:

```
git status
fatal: Not a git repository: my_old_path/.git/modules/astropy_helpers
```

After a while I discovered a Stackoverflow post, that explains how older verison of git (e.g. the one on my distribution) used absolute paths to refer to submodules. Thus, either you have to manually edit those files after moving your git repository, or, instead of `mv` the astropy directory `git clone` at the new location, then delete at the old location (This also makes it impossible to sync astropy development over different machines with dropbox as I used to do - each machine needs to have it's own clone unless the directory name is exactly the same).
Here is the Stackoverflow answer with more details:
http://stackoverflow.com/questions/10144149/git-status-returns-fatal-not-a-git-repository-but-git-exists-and-head-has-prop

Apparently, this is fixed in newer versions of git, but I still propose to at least mention it somewhere (If not in the documentation, then at least in this issue ;-)  ).

**Comment by jehturner** [comment]
This just caught me out somehow (can't even remember what I did to cause it). Re-cloning the repo from github and copying my changes into the new clone before committing/pushing did the trick...

**Comment by embray** [comment]
@nden Could this be what you were referring to this morning?  If so, it only has to do with moving entire repositories on the filesystem, and not with renaming submodules.  I'm not sure though.

**Comment by nden** [comment]
In my case, when I was setting up a package using `package-template` I must have typed the commands manually, so for quite a while I had `astropy-helpers` as submodule instead of `astropy_helpers`. I was seeing messages on `readthedocs` and `travis` like 
`astropy-helpers was not found. Did you forget to run git submodule add` but my brain never registered the difference. As I found, with older versions of git, it's not sufficient to remove the submodule and add the correct one. One has to manually change also `.gitmodules` and possibly `.git/config`. With later versions of git this is all done with one command:

`git mv old-submodule-name new-submodule-name`

I did it with v 2.x (read 1.8.x should be fine too).
Probably not worth mentioning in the docs since this is just one case (always count on me to do something like this). However, using a recent git version is important!
